### PR TITLE
feature: New IP parsing classes.  Removes usage of Pear Net_IPv4 and Net_IPv6.

### PR DESCRIPTION
**Pull request by murrant**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

**Comment by murrant** [comment]
Ready for review, sorry, I wasn't too happy with the initial implementation.  Feeling happy with it now.

**Comment by laf** [comment]
```
Array ( [0] => 1 [1] => Call to undefined function is_ip() [2] => /opt/librenms/html/includes/dev-overview-data.inc.php [3] => 35 )
```

That's device overview page.

File: html/includes/table/address-search.inc.php [hunk 1]
```diff
@@ -81,14 +83,12 @@
     $speed = humanspeed($interface['ifSpeed']);
     $type  = humanmedia($interface['ifType']);
 
-    if ($_POST['search_type'] == 'ipv6') {
-        list($prefix, $length) = explode('/', $interface['ipv6_network']);
-        $address               = Net_IPv6::compress($interface['ipv6_address']).'/'.$length;
+    if ($_POST['search_type'] == 'mac') {
```
Comment (laf) [inline html/includes/table/address-search.inc.php]: Think this is muddled up. Two checks for mac and no ipv4_network now.

**Comment by laf** [comment]
Going through a full discovery now, tested all the webui elements changed + api which are all fine except the one above.

Will see how polling is going after 15 minutes.

**Comment by laf** [comment]
```
Checking BGP peer [IPV6 Address here] PHP Fatal error:  Call to undefined method LibreNMS\Util\IPv6::compress() in /opt/librenms/includes/functions.php on line 787

Fatal error: Call to undefined method LibreNMS\Util\IPv6::compress() in /opt/librenms/includes/functions.php on line 787
```

**Comment by laf** [comment]
Discovery ran through ok. So just that poller error and that's it.

File: html/includes/table/address-search.inc.php (hunk 1 above)
Comment (murrant) [inline html/includes/table/address-search.inc.php]: shit, yeah, I started to change that, then changed it back (apparently only partially)

**Comment by murrant** [comment]
@laf I only was able to fully test discovery modules ipv4-addresses and ipv6-addresses.

Can you (or anyone) test bgp-peers, cisco-sla, loadbalancers/f5-ltm, ntp/cisco

**Comment by laf** [comment]
I could only test arista bgp. The rest are kit I don't have in my test lab.

**Comment by murrant** [comment]
@laf: I think the parts of bgp I touch are in a common code area.

**Comment by laf** [comment]
Running another full disco to test. After that I can't test anything more.

**Comment by laf** [comment]
Discovery completed.

**Pull request merged**